### DISCUSSION TITLE
DB Snapshot: Roles and Permissions

### DISCUSSION THREAD

**Discussion by pgegenfurtner** [directus/directus#16681] [Feature Requests]
When making a snapshot of the database roles and permissions do not get snapshotted.
This would be really useful, to use this snapshot function as a way to move from development to production.

Is this possible and a good idea?

An alternative I see right now, would be to build a custom db migration, right?

Thanks!

**Comment by br41nslug** [comment]
Snapshots only contain the Schema definition not the context. Users / Permissions contents can be exported using the API `GET /users?export=csv` and `GET /permissions?export=json` (Users can be exported in the App too but permissions not unfortunately)

https://docs.directus.io/reference/query.html#export

**Reply by pgegenfurtner** [reply]
Yeah, that makes sense to me.
How would I apply / import the exported permissions? 

Or is there generally a better way to track the migrations of my database / roles to move from dev to prod?

**Reply by br41nslug** [reply]
You'd need to use the import endpoint for that https://docs.directus.io/reference/system/utilities.html#import-data-from-file
Im not sure what the best way is for your use-case but there are a couple of ways you can approach this:
- some people prefer to stick to what they know and do SQL migrations that works
- we have migration scripts that can do the job (but lack cloud support)
- import/export endpoints (as i suggested above)
- you can ensure the environments are in sync over the API with an external tool/script
- or even build a custom extension to do it.

Specifically for roles/permissions i think migrations might be the way to go.

**Reply by pgegenfurtner** [reply]
Thank you so much! This is great advice I can work with

**Comment by paescuj** [comment]
I feel like this question has been coming up more and more lately, so this is a quick reminder to look up existing discussions before creating new ones 😃 I think it's more effective to upvote or comment on existing discussions so the team can easily see which topics are most actively discussed / wanted.

Relevant discussions from a quick search:
#16612
#11367
#15964
#15921
#10496
#9542

**Reply by pgegenfurtner** [reply]
Well. Obviously my search here sucked 😅
I looked in issues, not discussions.

Thank you very much!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by NilsBaumgartner1994** [comment]
Would wish to reopen this Feature

**Reply by rijkvanzanten** [reply]
Pretty sure https://github.com/directus/directus/discussions/13041 will solve the same use case 👍🏻
